prober/utils.go

```diff
@@ -63,5 +62,7 @@ func chooseProtocol(preferredIPProtocol string, target string, registry *prometh
 	}
 
 	level.Info(logger).Log("msg", "Resolved target address", "ip", ip)
-	return ip, nil
+	lookupTime := time.Since(resolveStart).Seconds()
+	probeDNSLookupTimeSeconds.Add(lookupTime)
```

Comment:
This is going to double count with the defer above. We should just have the defer, and can use a named return value for it.